### PR TITLE
feat: Introduce Comprehensive Rollout Compatibility Testing

### PR DESCRIPTION
**Pull request by dwertent**
**Closes:** #755 

## Overview

This pull request introduces a critical new capability to our CI/CD pipeline: a comprehensive, automated backward compatibility testing strategy. Previously, we had no automated way to validate software rollouts. This new workflow addresses that gap by simulating a full, sequential upgrade path, ensuring that a new release candidate is compatible with all previous final releases since `v0.10.0`. This significantly increases our confidence in the stability and reliability of the upgrade process.

### Key Changes

* **New Workflow** (`test-rollout-compatibility.yaml`): The core of this feature, this workflow dynamically discovers all prior release tags and orchestrates a three-phase test: caching data from old versions, upgrading to the new RC, and verifying data integrity.

* **Edge Case Test** (`test-v0.9-to-v0.10.yaml`): A dedicated workflow to handle the specific `v0.9.0` to `v0.10.0` upgrade, which falls outside the main compatibility test since the scripts are not in the `v0.9.0` tag.

* **Enhanced Test Runner** (`run-examples.sh`): The script is now capable of accepting a cache directory and version tag as command-line arguments, allowing it to act as a flexible test runner controlled by the workflow.

* **Consistent CRD Chart Releases**: Release the CRD chart alongside every operator chart release to ensure version alignment between the CRDs and the operator.

### CODE REVIEW

**Review by matthew1001 — COMMENTED** [review]
This will be really useful to have as part of the release process, thanks @dwertent Just a few comments

File: .github/workflows/test-rollout-compatibility.yaml [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+          tags_array=$(echo '${{ steps.get_tags.outputs.tags }}' | jq -r '.[]')
+          if [[ -z "$tags_array" ]]; then
+            echo "No previous tags found, skipping Phase 1"
+            exit 0
+          fi
+          
+          # This loop simulates a real-world cluster being upgraded over time.
+          # For each historical version, we install it, check out its corresponding source code,
+          # and run the examples to generate and cache state data.
+          for tag in $tags_array; do
```
[the remark this answers is not in the document]
Comment (dwertent) [reply .github/workflows/test-rollout-compatibility.yaml:124]: We first run `./scripts/default-installation-validation.sh paladin`, which performs a basic installation check — confirming that the pods are running with the expected image and that components like `scd`, `reg`, etc., are deployed to the chain (by inspecting the RCs).

Next, we execute:

```bash
RUN_COMMANDS=start PALADIN_ABI_VERSION="$tag" ./scripts/run-examples.sh "${{ steps.setup_cache_directory.outputs.CACHE_BASE_DIR }}" "$tag"
```

This tests Paladin’s functionality by ensuring the examples run successfully.

Finally, we validate that the addresses remain present and accessible.


I did my best to avoid false-positives, however, since we are running on GH runners, there is always the chance that it will fail.

File: examples/public-storage/src/tests/data-persistence.ts [hunk 2]
```diff
@@ -0,0 +1,247 @@
+import PaladinClient from "@lfdecentralizedtrust-labs/paladin-sdk";
+import storageJson from "../abis/Storage.json";
```
[the remark this answers is not in the document]
Comment (matthew1001) [reply examples/public-storage/src/tests/data-persistence.ts:16]: OK, I can see why they ended up there. I don't know if it's worth adding a readme to the `/tests/` folders to explain their purpose, but it seems OK to have them there for now. And moving them later if others feel strongly isn't a big job

[the remark this answers is not in the document]
Comment (dwertent) [reply examples/public-storage/src/tests/data-persistence.ts:16]: I added the [develop.md](https://github.com/LF-Decentralized-Trust-labs/paladin/pull/763/files#diff-57819195e24fbd71bd887fc279726f3e7d4ed4bb0d569872f483b3cf0a29c8bb) file for this

File: .github/workflows/test-rollout-compatibility.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dwertent) [reply .github/workflows/test-rollout-compatibility.yaml:125]: > So we'll drop out with a failed GHA if they don't run successfully?

Yes
